game.exs

```diff
@@ -1,7 +1,58 @@
 defmodule GuessingGame do
 
-  def hi() do
-    IO.puts("hello world")
+  def compare(n, t) do
+    # Best Practice ?
+
+    # a)
+    # case n do
+    #   m when m < t -> :low
+    #   m when m == t -> :equal
+    #   m when m > t -> :high
+    # end
+
+    # b)
+    cond do
+      n < t -> :low
+      n == t -> :equal
+      n > t -> :high
+    end
+  end
+
+  def print_state(state) do
+    case state do
+      :low -> IO.puts("Too low")
+      :equal -> IO.puts("That's the right number, you win!")
+      :high -> IO.puts("Too high")
+    end
+  end
+
+  def print_turns(turns) do
+    IO.puts("You have #{turns} turns left")
+    if (turns == 0) do
+      IO.puts("You lose")
+    end
+  end
+
+  def get_guess() do
+    IO.gets("Guess a number: ") |> String.trim |> String.to_integer
+  end
+
+  def play() do
+    main(Enum.random(1..100), 7)
   end
 
+  def turn(number) do
```

Comment:
Why don't you actually pull `turn` into `main`, we may break it out again but I think it might allow us to look at things a little differently.